spec/GraphQLOverHTTP.md

```diff
@@ -537,17 +537,34 @@ The following examples provide guidance on how to deal with specific error cases
 when using the `application/graphql+json` media type to encode the response
 body:
 
-##### Invalid parameters or parsing failure
+##### JSON parsing failure
 
-For example a POST response body of `NONSENSE` or `{"qeury": "{__typena`.
+For example a POST response body of `NONSENSE` or `{"query":` (note: invalid
+JSON).
 
-Requests that do not constitute a well-formed GraphQL request should result in
-status code `400` (Bad Request).
+Requests that the server cannot interpret should result in status code `400`
+(Bad Request).
+
+##### Invalid parameters
+
+For example a POST response body of `{"qeury": "{__typename}"}` (note: typo) or
+`{"query": "query Q ($i:Int!) { q(i: $i) }", "variables": [7]}` (note: invalid
+shape for `variables`).
+
+Requests that do not constitute a well-formed _GraphQL-over-HTTP request_ should
+result in status code `400` (Bad Request).
+
+##### Document parsing failure
+
+For example a POST response body of `{"query": "{"}`.
+
+Requests where the _GraphQL document_ cannot be parsed should result in status
+code `400` (Bad Request).
 
 ##### Document validation failure
 
-Requests that fail to pass validation SHOULD be denied execution with a status
-code of `400` (Bad Request).
+Requests that fail to pass _GraphQL validation_ SHOULD be denied execution with
+a status code of `400` (Bad Request).
```

Comment:
Agreed.  Is there a difference between `should result in status code '400' (Bad Request).` and `SHOULD be denied execution with a status code of '400' (Bad Request)`?

Reply:
In my opinion this is concretely different from the others - you _can_ execute this one, you cannot execute the others (because they weren't parseable so you could not determine what to execute. And so the denial of execution is what you're doing (you can't deny execution to something that you couldn't have executed in the first place).

However, "result in" would also cover it, so if you both feel strongly that wording is better/clearer I'd be happy to make the change.

Reply:
I can see the difference, how about a note that explains it?

Reply:
I'm not sure what to put in the note.

Reply:
How about:

> Note: In this case, it would be feasible for the server to attempt execution despite validation errors.
> Although generally not recommended, exceptions might be necessary to support legacy clients.